notebooks/introductory_infilling.ipynb

```diff
@@ -0,0 +1,2572 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Introduction to Crunchers\n",
+    "\n",
+    "In this notebook we demonstrate crunchers, silicone's most fundamental infilling methods. You will need to install silicone from as described in the readme in order to run this code. It introduces two specific crunchers and the basic tools to manipulate them. These two are the 'closest RMS' and the 'rolling window quantile' crunchers. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/javascript": [
+       "IPython.OutputArea.prototype._should_scroll = function(lines) { return false; }"
+      ],
+      "text/plain": [
+       "<IPython.core.display.Javascript object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "pyam - INFO: Running in a notebook, setting `pyam` logging level to `logging.INFO` and adding stderr handler\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os.path\n",
+    "import traceback\n",
+    "\n",
+    "import pandas as pd\n",
+    "import pyam\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "import silicone.database_crunchers\n",
+    "from silicone.utils import (\n",
+    "    _get_unit_of_variable,\n",
+    "    find_matching_scenarios,\n",
+    "    _make_interpolator,\n",
+    "    _make_wide_db, \n",
+    "    get_sr15_scenarios\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Assembling example data\n",
+    "\n",
+    "Here we pull some example data by downloading a selection of the SR1.5 scenarios from the IIASA database. If the data has already been downloaded before, we will use that instead for brevity. We select only a few cases"
```

Comment:
Before this will be published, it would be good to provide a reference for the data, e.g. 

Daniel Huppmann, Elmar Kriegler, Volker Krey, Keywan Riahi, Joeri Rogelj, Katherine Calvin, Florian Humpenoeder, Alexander Popp, Steven K. Rose, John Weyant, Nico Bauer, Christoph Bertram, Valentina Bosetti, Jonathan Doelman, Laurent Drouet, Johannes Emmerling, Stefan Frank, Shinichiro Fujimori, David Gernaat, Arnulf Grubler, Celine Guivarch, Martin Haigh, Christian Holz, Gokul Iyer, Etsushi Kato, Kimon Keramidas, Alban Kitous, Florian Leblanc, Jing-Yu Liu, Konstantin Löffler, Gunnar Luderer, Adriana Marcucci, David McCollum, Silvana Mima, Ronald D. Sands, Fuminori Sano, Jessica Strefler, Junichi Tsutsui, Detlef Van Vuuren, Zoi Vrontisi, Marshall Wise, and Runsen Zhang.
IAMC 1.5°C Scenario Explorer and Data hosted by IIASA.
Integrated Assessment Modeling Consortium & International Institute for Applied Systems Analysis, 2019.
doi: 10.5281/zenodo.3363345 | url: data.ene.iiasa.ac.at/iamc-1.5c-explorer

Reply:
OK, I've made it shorter but there's a citation